### PR TITLE
[REF TEST] Refactor and add unit tests for `check_dependency`

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR proposes to make a few changes to `clinica/utils/check_dependency.py`:

- Add type hints to functions
- Update existing docstrings to follow Numpydoc conventions
- Add documentations and examples
- Refactor check functions to be partials and not duplicate the checking logic
- Add unit tests
- Remove `check_software_requirements` which was not used

### CODE REVIEW

File: test/unittests/utils/test_check_dependency.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        ClinicaMissingDependencyError,
+        match="The FOO environment variable you gave is not a folder",
+    ):
+        check_environment_variable("FOO", "foo")
+    os.environ["FOO"] = "."
+    assert check_environment_variable("FOO", "foo") == "."
+    os.environ.pop("FOO")
+
+
+def test_check_software():
```
Comment (MatthieuJoulot) [inline test/unittests/utils/test_check_dependency.py:33]: It feels like this function could easily be factorized in two functions using pytest. One would test the errors, while the other tests the assertions. WDYT ?

**Review by MatthieuJoulot — CHANGES_REQUESTED** [review]
Just one small change, otherwise this looks good ! 
One minor question: should we do the same thing you're doing for FSL for Freesurfer, and check that the version is recent enough ? it would make sense to have only versions 6.0 and above (since that is what we handle with `freesurfe-atlas`

File: test/unittests/utils/test_check_dependency.py (hunk 1 above)
Comment (NicolasGensollen) [reply test/unittests/utils/test_check_dependency.py]: I proposed a possible refactoring in cdf8f93
Let me know if this is more or less what you had in mind.

**Comment by NicolasGensollen** [comment]
> One minor question: should we do the same thing you're doing for FSL for Freesurfer, and check that the version is recent enough ? it would make sense to have only versions 6.0 and above (since that is what we handle with `freesurfe-atlas`

I added a version check in the same spirit in cdf8f93. Let me know if this works for you.

**Review by MatthieuJoulot — APPROVED** [review]
Looks good :)

**Pull request merged**